lib/rules/no-assignment-of-computed-property-dependencies.js

```diff
@@ -0,0 +1,196 @@
+ */
+function getComputedPropertyDependentKeys(node) {
+  if (!node.arguments) {
+    return [];
+  }
+
+  return node.arguments
+    .filter((arg) => arg.type === 'Literal' && typeof arg.value === 'string')
+    .map((node) => node.value);
+}
+
+/**
+ * Gets a list of computed property dependency keys used inside a class.
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of dependent keys used inside the class
+ */
+function findComputedPropertyDependentKeys(nodeClass) {
+  if (types.isClassDeclaration(nodeClass)) {
+    // Native JS class.
+    return javascriptUtils.flatMap(nodeClass.body.body, (node) => {
+      const computedDecorator = decoratorUtils.findDecorator(node, 'computed');
+      if (computedDecorator) {
+        return getComputedPropertyDependentKeys(computedDecorator.expression);
+      } else {
+        return [];
+      }
+    });
+  } else if (types.isCallExpression(nodeClass)) {
+    // Classic class.
+    if (
+      nodeClass.arguments &&
+      nodeClass.arguments.length > 0 &&
+      types.isObjectExpression(nodeClass.arguments[nodeClass.arguments.length - 1])
+    ) {
+      const classObject = nodeClass.arguments[nodeClass.arguments.length - 1];
+      return javascriptUtils.flatMap(classObject.properties, (node) => {
+        if (emberUtils.isComputedProp(node.value) && node.value.arguments) {
+          return getComputedPropertyDependentKeys(node.value);
+        } else {
+          return [];
+        }
+      });
+    }
+  } else {
+    assert(false, 'Unexpected node type for an Ember class.');
+  }
+
+  return [];
+}
+
+/**
+ * Gets a list of tracked properties used inside a class.
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of tracked properties used inside the class
+ */
+function findTrackedProperties(nodeClassDeclaration) {
+  return nodeClassDeclaration.body.body
+    .filter(
+      (node) =>
+        types.isClassProperty(node) &&
+        decoratorUtils.hasDecorator(node, 'tracked') &&
+        types.isIdentifier(node.key)
+    )
+    .map((node) => node.key.name);
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'disallow assignment of untracked properties that are used as computed property dependencies',
+      category: 'Computed Properties',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-assignment-of-computed-property-dependencies.md',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    if (emberUtils.isTestFile(context.getFilename())) {
+      // This rule does not apply to test files.
+      return {};
+    }
+
+    // State being tracked for this file.
+    let trackedProperties = undefined;
+    let computedPropertyDependentKeys = undefined;
+    let currentEmberModule = null;
```

Comment:
This variable name is odd, what does it represent?

Reply:
The node for the current Ember module (component, controller, etc) that we're inside. Made some slight updates to naming/comments for clarity.